### ISSUE TITLE
Switch label

### ISSUE THREAD

**Issue by andresz1** [adevinta/spark#535]
### Description

The switch component should have a label integrated

### Problem Statement/Justification

The switch component is missing a label and it's needed in terms of accessibility. ATM could be added by the user but to provide a better DX I think it should be provided out of the box. 

### Proposed Solution or API

```tsx
<Switch> {/* label */}
  <SwitchInput /> {/* input */}

  <SwitchText> {/* span */}
     Groceries
  </SwitchText>
</Switch>
```
